src/tools/miri/tests/pass/function_pointers.rs

```diff
@@ -75,7 +75,6 @@ fn main() {
     let g = f as fn() -> i32;
     assert!(return_fn_ptr(g) == g);
     assert!(return_fn_ptr(g) as unsafe fn() -> i32 == g as fn() -> i32 as unsafe fn() -> i32);
-    assert!(return_fn_ptr(f) != f);
```

Comment:
Let's leave a comment for next time... I would have made this a suggestions but github doesn't let me
```
// Check that `g` always has the same address. This is not guaranteed
// but Miri currently uses a fixed address for monomorphic functions.
(the two lines checking `g`)
// We don't check anything for `f`. Miri gives it many different addresses
// but mir-opts can turn them into the same address.
let _val = return_fn_ptr(f) != f;
```